src/gt4py/stencil_object.py

```diff
@@ -245,9 +245,11 @@ def _make_origin_dict(origin: Any) -> Dict[str, Index]:
 
         raise ValueError("Invalid 'origin' value ({})".format(origin))
 
+    @staticmethod
     def _get_max_domain(
```

Comment:
Is my understanding correct that the reason for making this static is that it's used in `normalize_args` which is used in `__sdfg__`, and that dunder call needs to be static?

Reply:
If that's correct, I guess a follow-up question is: does this need to change if we only need them static in the `DaceStencilObject`, since they're not part of the public interface of the object?

Reply:
`__sdfg__` doesn't have to be static, but the `__sdfg__` method of DaCeLazyStencil can't access the `self.implementation` in `__sdfg__`, since that would trigger a build. So we _must_  access `normalize_args` through the type instead of the instance.